src/geb/trans.lisp

```diff
@@ -78,3 +81,77 @@
   "Turns a @GEB-SUBSTMORPH to a Vamp-IR Term"
   (assure list
     (geb.poly:to-circuit (to-poly obj) name)))
+
+
+
+
+
+
+(defmethod to-bitc ((obj <substobj>))
+  (typecase-of substobj obj
+    (so0     0)
+    (so1     0)
+    (coprod  (+ 1 (max (to-bitc (mcar obj)) (to-bitc (mcadr obj)))))
+    (prod    (+ (to-bitc (mcar obj)) (to-bitc (mcadr obj))))
+    (otherwise (subclass-responsibility obj))))
```

Comment:
Just a question, if we see a `<substobj>` in a `morphism` slot then it's the identity, is this saved at all?

I know for things that take `objects` this is correct

For example
```lisp
(comp so1 so1)
```

is just `identity . identity` for the type so1

Reply:
I don't know what you mean by "saved". to-bitc, when applied to a substobj, just calculates the bit-width required to store the object (in other words, it converts a GEB object into a BITC object). It's generally used when calculating the bitwidth of a domain. Although, it may not be necessary. `(to-bitc (dom X))` should be the same as `(dom (to-bitc X))`, etc.

Reply:
I wasn't aware that objects become identities geb morphism slots. I did not write any code with that in mind.

Reply:
might be good to update the code with that if possible

Reply:
Alright, I think I've fixed that.